### PR TITLE
GNOME: 3.38 → 40

### LINKED ISSUE #116252: Tracking GNOME 40 update
Release candidate has been released and 40.0 [planned](https://wiki.gnome.org/Forty) for March 24.

- Project: https://github.com/NixOS/nixpkgs/projects/40
- Previous bump: https://github.com/NixOS/nixpkgs/issues/95248

### Important considerations
- Shell 40 will [not support](https://blogs.gnome.org/shell-dev/2021/03/12/gnome-40-your-extension/) extensions not explicitly marked as compatible.

cc @NixOS/gnome

### PR DESCRIPTION
**Pull request by jtojnar**
https://help.gnome.org/misc/release-notes/40.0/
https://forty.gnome.org/

Fixes: https://github.com/NixOS/nixpkgs/issues/116252

**The changes are reasonably tested but there might be [a regression with fingerprint readers](https://github.com/NixOS/nixpkgs/pull/117081#issuecomment-828232633). If you have one, we would appreciate if you could [help debug it](https://github.com/NixOS/nixpkgs/pull/117081#issuecomment-828356484).**

~~Currently, mostly update script changes and updates produced by it are included. Feel free to test packages, send fixup commits against the branch or [read changelogs](https://github.com/jtojnar/what-changed) and point out possible problems.~~

- [x] gtkmm, glibmm, libsigcxx, libsoup, gtksourceview, libgda, lasem are new parallel-installable series ~~so do not bump them for now.~~

https://hydra.nixos.org/jobset/nixpkgs/gnome#tabs-evaluations

## How to test

1. Clone the nixpkgs repo and checkout the `gnome-40` branch
2. Save the following NixOS configuration file
<details><summary>configuration.nix</summary>

```nix
{ pkgs, config, ... }: {
  environment.systemPackages = with pkgs; [
    gdb
    binutils # readelf
    file
    htop
    less
    git
    rr
    gnomeExtensions.gsconnect
    dfeet
  ];
  environment.enableDebugInfo = true;
  services.xserver = {
    enable = true;
    layout = "cz";
    xkbVariant = "qwerty";

    displayManager.gdm = {
      enable = true;
      debug = true;
    };

    desktopManager.xterm.enable = false;
    desktopManager.gnome3 = {
      enable = true;
      debug = true;
      flashback = {
        enableMetacity = true;
      };
    };
  };

  services.gnome3.core-developer-tools.enable = true;
  services.gnome3.games.enable = true;
  services.openssh.enable = true;
  programs.gpaste.enable = true;
  users.mutableUsers = false;

  users.extraUsers.jtojnar = {
    isNormalUser = true;
    uid = 1000;
    extraGroups = [ "wheel" "networkmanager" ];
    password = "holandaise";
    openssh.authorizedKeys.keys = ["ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDYbOlZydfRRCGCT08wdtPcpfSrgxMc6weDx3NcWrnMpVgxnMs3HozzkaS/hbcZUocn7XbCOyaxEd1O8Fuaw4JXpUBcMetpPXkQC+bZHQ3YsZZyzVgCXFPRF88QQj0nR7YVE1AeAifjk3TCODstTxit868V1639/TVIi5y5fC0/VbYG2Lt4AadNH67bRv8YiO3iTsHQoZPKD1nxA7yANHCuw38bGTHRhsxeVD+72ThbsYSZeA9dBrzACpEdnwyXclaoyIOnKdN224tu4+4ytgH/vH/uoUfL8SmzzIDvwZ4Ba2yHhZHs5iwsVjTvLe7jjE6I1u8qY7X8ofnanfNcsmz/ jtojnar@kaiser"];
  };
  nixpkgs.config.allowAliases = false;

  virtualisation.qemu.options = [ "-device intel-hda -device hda-duplex" ];
  virtualisation.memorySize = 2048;
  virtualisation.diskSize = 8048;
}
```

</details>

3. Run the following command (do not forget to adjust the path to the nixpkgs checkout and the configuration): 
	- On NixOS: `env NIX_PATH=nixpkgs=$HOME/Projects/nixpkgs:nixos-config=path/to/configuration.nix nixos-rebuild build-vm`
	- Outside NixOS: `env NIX_PATH=nixpkgs=$HOME/Projects/nixpkgs:nixos-config=path/to/configuration.nix nix-shell -p '(nixos {}).nixos-rebuild' --run 'nixos-rebuild build-vm'`

### CODE REVIEW

File: pkgs/desktops/gnome-3/core/baobab/default.nix
Comment (SuperSandro2000) [inline pkgs/desktops/gnome-3/core/baobab/default.nix:13]: You missed that hash.

File: pkgs/desktops/gnome-3/apps/gedit/default.nix
Comment (SuperSandro2000) [inline pkgs/desktops/gnome-3/apps/gedit/default.nix:30]: And you missed that hash.

File: pkgs/data/documentation/gnome-user-docs/default.nix
Comment (SuperSandro2000) [inline pkgs/data/documentation/gnome-user-docs/default.nix:16]: And you missed that hash.

File: pkgs/desktops/gnome-3/core/baobab/default.nix
Comment (jtojnar) [reply pkgs/desktops/gnome-3/core/baobab/default.nix:13]: For some reason the servers are timing out for me.

File: pkgs/data/documentation/gnome-user-docs/default.nix
Comment (dotlambda) [reply pkgs/data/documentation/gnome-user-docs/default.nix:16]: and the commit message is wrong

**Comment by symphorien** [comment]
it looks like geocode-glib is incompatible with the new version of libsoup
```
Project name: geocode-glib
Project version: 3.26.2
C compiler for the host machine: gcc (gcc 10.2.0 "gcc (GCC) 10.2.0")
C linker for the host machine: gcc ld.bfd 2.35.1
Host machine cpu family: x86_64
Host machine cpu: x86_64
Configuring config.h using configuration
meson.build:29: WARNING: Consider using the built-in warning_level option instead of using "-Wall".
Program glib-mkenums found: YES (/nix/store/a9j3wfymvw7bd2wygg9qz7m4xriican5-glib-2.68.0-dev/bin/glib-mkenums)
Found pkg-config: /nix/store/i7sfbrfn7y5dhgd2k19ys0gv2scz64a1-pkg-config-wrapper-0.29.2/bin/pkg-config (0.29.2)
Run-time dependency gio-2.0 found: YES 2.68.0
Run-time dependency json-glib-1.0 found: YES 1.6.2
Did not find CMake 'cmake'
Found CMake: NO
Run-time dependency libsoup-2.4 found: NO (tried pkgconfig)

geocode-glib/meson.build:36:0: ERROR: Dependency "libsoup-2.4" not found, tried pkgconfig
```

**Comment by jtojnar** [comment]
Yeah, another big bump I forgot about. Reverted for now.

**Comment by maxeaubrey** [comment]
> gtkmm, glibmm, libsigcxx, libsoup are new parallel-installable series so do not bump them for now.

Also atkmm and pangomm.

**Comment by maxeaubrey** [comment]
`gtksourceview` is a major bump as well, depending on gtk4. We should stick with 4.8 according to upstream.

**Comment by worldofpeace** [comment]
I had graham switch the branch for the jobset :+1:

**Comment by jtojnar** [comment]
Just finished reviewing the changelogs. Looks mostly good, added the issues I noticed into the project: https://github.com/NixOS/nixpkgs/projects/40

**Comment by SuperSandro2000** [comment]
It got brought to my attention that many extensions will break with gnome 40 like https://github.com/NixOS/nixpkgs/blob/master/pkgs/desktops/gnome-3/extensions/dash-to-panel/default.nix. How will that be handled by the update?

**Comment by jtojnar** [comment]
Extensions that are incompatible with GNOME 40 will be automatically disabled at runtime.

**Comment by jtojnar** [comment]
Updated few more packages and rebased to resolve the conflicts.

**Comment by jtojnar** [comment]
libhandy no longer depends on webkitgtk through glade.

Confirmed it works using `env GLADE_CATALOG_SEARCH_PATH=$(nix-build -A libhandy.glade --no-out-link)/share/glade/catalogs GLADE_MODULE_SEARCH_PATH=$(nix-build -A libhandy.glade --no-out-link)/lib/glade/modules $(nix-build -A glade --no-out-link)/bin/glade`.

**Comment by jtojnar** [comment]
Rebased, hopefully for the last time. I tested it somewhat and I believe @maxeaubrey have been also using it so I would consider it good to go once hydra chews through the jobset.

**Comment by maxeaubrey** [comment]
Can confirm, I've been running it as a daily driver on my personal laptop for the last week without much issue - I had one non-reproducible instance of logging in taking a very long time, but without anything in the logs, so... :woman_shrugging: 

We might want to revert librsvg and bump it to 2.50.4 instead? Right now it's on unstable, and afaict 2.50.x is still used for the GNOME 40 release, based on build-meta.

**Comment by jtojnar** [comment]
Thanks for pointing out the librsvg issue. I waited for the stable version to materialize but it never did. Guess it is more rebuilds 🤷‍♀️

Checked all the packages whose version matches `\.\d*[13579]\.|alpha|beta|rc|\.9` to see whether it should be safe to use them:

- tracker – stable
- tracker-miners – stable
- grilo-plugins – stable
- amtk – stable
- folks – unstable, important clean ups
- gnome-multi-writer – rc, only a bugfix
- gnome-online-accounts – rc, only few bugfixes
- gnome3.geary – rc, looks stable enough, important fixes
- vte – rc
- gthumb – stable?
- gnome-calendar – rc, as stable as it gets
- adwaita-icon-theme – rc
- gnome3.file-roller – alpha, fixes cve

**Comment by maxeaubrey** [comment]
`vte` also has 0.64.0. I'll open a PR fixing that and librsvg in a bit.

**Comment by jtojnar** [comment]
> `vte` also has 0.64.0.

gnome-calendar too but they did not push tarballs.

> I'll open a PR fixing that and librsvg in a bit.

 Already done with the rebase.

**Comment by Abdillah** [comment]
In my case, there still some error,

- the gnome-shell missing EGL header (`mesa` dependency fixed this)
- metacity missing `xorg.libXdestroy`
- gnome-flashback missing `xorg.libXxf86vm`

And I'm wondering on how to make the animation smoother in Qemu? It seems increasing the `memsize` doesn't help.

**Comment by jtojnar** [comment]
Since staging-next has been merged, we can target this to master now. cc @vcunat @FRidh

**Comment by jonringer** [comment]
@GrahamcOfBorg eval

**Comment by jonringer** [comment]
if the rebuild is correct, this should not be targeting master.

**Comment by jtojnar** [comment]
@jonringer It has already been built by Hydra: https://hydra.nixos.org/jobset/nixpkgs/gnome#tabs-evaluations

**Comment by jonringer** [comment]
Ah, wasn't aware. Sounds good then :)

**Comment by tenaf0** [comment]
I get an occasional error with programs that try to open the gtk file chooser dialog. It happens with programs built on non-gnome-40 branch like firefox, chromium and krita and it is fixed by building it on gnome-40. So it's not a show-stopper, just wanted to mention it here.

Here is the error message in case of krita:
```
(.krita-wrapped:19377): Gtk-WARNING **: 12:19:55.482: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.
**
Gtk:ERROR:../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /nix/store/c20nvqh5cbzwbi5jcmz8agd6xw7glsji-adwaita-icon-theme-40.0/share/icons/Adwaita/16x16/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
Bail out! Gtk:ERROR:../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /nix/store/c20nvqh5cbzwbi5jcmz8agd6xw7glsji-adwaita-icon-theme-40.0/share/icons/Adwaita/16x16/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
Aborted (core dumped)
```

And in case of chromium:
```
Gtk:ERROR:../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /nix/store/24xvj3dmqpxv1ypa7wsj7iy937s6x5xh-adwaita-icon-theme-3.38.0/share/icons/Adwaita/16x16/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
Bail out! Gtk:ERROR:../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /nix/store/24xvj3dmqpxv1ypa7wsj7iy937s6x5xh-adwaita-icon-theme-3.38.0/share/icons/Adwaita/16x16/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
Aborted (core dumped)
```

**Comment by dasj19** [comment]
I will try this in a VM... what is left to be tested? ... I could try my usual gnome setup that I have in configuration.nix

**Comment by jtojnar** [comment]
Me and @ maxeaubrey have been running it on our production systems without a hitch for a while now so I would consider this ready to merge. Of course more testing is always welcome.

There are still some [failing builds](https://hydra.nixos.org/eval/1665478#tabs-still-fail), a few due to the new GLib but those are mostly leaf packages so I would not bother much with them.

**Comment by dasj19** [comment]
So far I found an error in file-roller.
Steps to reproduce:
1. Download the zip file of gnome-40 branch from github: https://github.com/NixOS/nixpkgs/archive/refs/heads/gnome-40.zip
2. Open the archive with Archive Manager (file-roller)
3. Notice the failure to unzip: `An error occured while extracting files Fatal error`
Workaround: Right click on the archive and then extract.

**Comment by jtojnar** [comment]
I am pretty sure I have seen that in 3.38 too.

**Comment by vcunat** [comment]
I see many darwin regressions due to librsvg not building: https://hydra.nixos.org/build/141650131

**Comment by dasj19** [comment]
@jtojnar I tried a couple of things with file-roller:
1. Running `xnn5k0i84b7003q3l5qa18fgvh1zrrll-file-roller-3.38.0` from the unstable branch works well with that particular .zip file
2. If I install `unzip` it works well with the file-roller from gnome-40 branch
3. When running from console the following appears: `(file-roller:6709): GLib-GIO-CRITICAL **: 08:17:55.625: g_output_stream_write_all: assertion 'buffer != NULL' failed`

All of theese point findings point to a regression

**Comment by jtojnar** [comment]
I dropped the file-roller 3.39.1 commit to avoid conflict with 3.38.1 from
master. Could you please check if that version is okay?

File: pkgs/development/libraries/librsvg/default.nix
Comment (jtojnar) [inline pkgs/development/libraries/librsvg/default.nix:39]: > I see many darwin regressions due to librsvg not building: https://hydra.nixos.org/build/141650131

 Maybe adding Foundation here would fix that?

File: pkgs/development/libraries/librsvg/default.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/librsvg/default.nix:38]: darwin shouldn't be used as an input.

Comment (maralorn) [reply pkgs/development/libraries/librsvg/default.nix:38]: Can you explain what harm it does?

Comment (hmenke) [reply pkgs/development/libraries/librsvg/default.nix:38]: No harm per se, but the same reason you don't use `pkgs` as an input. Instead use `ApplicationServices` as an input directly.

Comment (SuperSandro2000) [reply pkgs/development/libraries/librsvg/default.nix:38]: > Can you explain what harm it does?

It is harder to override it.

**Comment by dasj19** [comment]
> I dropped the file-roller 3.39.1 commit to avoid conflict with 3.38.1 from master. Could you please check if that version is okay?
Unfortunately it didn't help. It helps having `unzip` or `p7zip` installed on the system.
It seems that the issue is much deeper in gnome libraries... and you reported it yourself here: https://gitlab.gnome.org/GNOME/file-roller/-/issues/40 ... I'm not knowledgeable enough but it might have roots in glib https://gitlab.gnome.org/GNOME/glib/-/issues/249
With that being said I do not know how to fix this issue.

**Comment by dasj19** [comment]
Another minor setback is the configuration over here: https://github.com/NixOS/nixpkgs/pull/67389#issuecomment-524565602 no longer works. The xdg folders do not get updated when changing it on a live system.

File: pkgs/development/libraries/librsvg/default.nix
Comment (jtojnar) [reply pkgs/development/libraries/librsvg/default.nix]: Done. Could you please check if adding the Foundation was sufficient?

**Comment by jtojnar** [comment]
As a stop gap, we could add unzip to PATH of file roller?

**Comment by dasj19** [comment]
> As a stop gap, we could add unzip to PATH of file roller?

Yes, that seems to help. While we are at it we should also add `cpio` to `buildInputs` because it says in the build output that it misses it.

**Comment by jtojnar** [comment]
Feel free to push that here while I try to set up https://github.com/fastai/fastmac.

File: pkgs/development/libraries/librsvg/default.nix
Comment (vcunat) [reply pkgs/development/libraries/librsvg/default.nix:41]: Bad luck, another issue: https://hydra.nixos.org/build/142180721/nixlog/1/tail

It's hard to do this without some access to a darwin machine

**Comment by jtojnar** [comment]
Okay, now it fails with less clear error:

```
    Finished release [optimized + debuginfo] target(s) in 5m 01s
/nix/store/14s1hh53dr8xkvxq3vs5rm9i6nnk4kar-cctools-port-949.0.1/bin/ranlib: file: .libs/librsvg_c_api.a(_rsvg_dummy.o) has no symbols
ranlib: file: .libs/librsvg_c_api.a(_rsvg_dummy.o) has no symbols
warning: ranlib: warning for library: .libs/librsvg_c_api.a the table of contents is empty (no object file members in the library define global symbols)
  CCLD     librsvg-2.la
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
warning: nm: no name list
  CCLD     rsvg-convert
  GISCAN   Rsvg-2.0.gir
  GICOMP   Rsvg-2.0.gir
 VAPIGEN librsvg-2.0.vapi
GdkPixbuf-2.0.gir:6346.55-6346.55: error: The symbol `GModule' could not be found
make[2]: *** [/nix/store/zc1llkjhjwqz9nv9a50a8hyjpqdpw21a-vala-0.52.2/share/vala/Makefile.vapigen:61: librsvg-2.0.vapi] Error 1
make[2]: Leaving directory '/private/tmp/nix-build-librsvg-2.50.4.drv-0/librsvg-2.50.4'
make[1]: *** [Makefile:1276: all-recursive] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-librsvg-2.50.4.drv-0/librsvg-2.50.4'
make: *** [Makefile:827: all] Error 2
builder for '/nix/store/6ph2scihzblq2pd0h86aicjvjzd3nrnx-librsvg-2.50.4.drv' failed with exit code 2
```

**Comment by jtojnar** [comment]
> Another minor setback is the configuration over here: [#67389 (comment)](https://github.com/NixOS/nixpkgs/pull/67389#issuecomment-524565602) no longer works. The xdg folders do not get updated when changing it on a live system.

I do not think it ever worked that way. It probably only has effect on login (IIRC some GNOME component is asking user to confirm that and then stores the values to `~/.local/share` overriding the config.)

**Comment by jtojnar** [comment]
@vcunat I disabled building the VAPI for now but not sure if something does not depend on that.

**Comment by vcunat** [comment]
Now [it builds](https://hydra.nixos.org/eval/1665972#tabs-now-succeed), so it's certainly better without vapi than not at all.

**Comment by jtojnar** [comment]
Fixed Pantheon DE and some other packages and it looks like the only remaining potential blocker is `libsForQt5.qtwebkit`. cc @periklis @abbradar @NixOS/qt-kde

**Comment by vcunat** [comment]
Speaking of blockers, `nixpkgs-unstable` channel would still be blocked due to `imagemagick.x86_64-darwin`: https://hydra.nixos.org/build/142193914#tabs-summary
(but the webkit issue seems more serious _to me_; I'm not a mac fan really)

**Comment by LEXUGE** [comment]
Might be an unrelated issue, but I tested out the GNOME 40 with fprintd enabled. The building worked fine, however, the login process was unsuccessful both in GDM and lock screen. If I type a few words in the password box, the box "shaked" with prompt like "Fingerprind didn't match, please try again".

**Comment by jtojnar** [comment]
qtwebkit is the old unmaintained project qtwebview is the new one IIRC so maybe it is not that bad.

**Comment by jtojnar** [comment]
Yeah, adding Foundation seems to have fixed imagemagick.

This one looks gnarlier: https://hydra.nixos.org/build/142193912/nixlog/1

**Comment by jtojnar** [comment]
> If I type a few words in the password box, the box "shaked" with prompt like "Fingerprind didn't match, please try again".

Do you mean that neither fingerprint nor regular password worked?

**Comment by LEXUGE** [comment]
> > If I type a few words in the password box, the box "shaked" with prompt like "Fingerprind didn't match, please try again".
> 
> @LEXUGE Do you mean that neither fingerprint nor regular password worked?

Yes, in other words, I have to rollback to login in.

**Comment by jtojnar** [comment]
That is quite serious. Could you please try enabling debugging with

```nix
services.xserver.displayManager.gdm.debug = true;
services.xserver.desktopManager.gnome3.debug = true;
```

and check the systemd journal (e.g. in TTY or over SSH)?

**Comment by LEXUGE** [comment]
@jtojnar Would you like to point out journals of what units/services should I be looking at? I will test it out as soon as I am available.

**Comment by jtojnar** [comment]
IIRC it might not be associated with any unit. I would look for anything with gdm/gnome-shell/gnome-session/pam/gcr/fprintd.

**Comment by dasj19** [comment]
I don't know if this is relevant... but if I setup a keyboard layout via Gnome Settings it does not apply to the gdm login screen which still uses en_US. There is no locale-related settings in my configuration.nix

**Comment by davidak** [comment]
>the gdm login screen which still uses en_US. There is no locale-related settings in my configuration.nix

@dasj19  `en_US` is the default. you probably have to use `i18n.defaultLocale = "en_US.UTF-8";` to change it globally

**Comment by davidak** [comment]
>GNOME 40

>`desktopManager.gnome3`

i argued in the iso name discussion that the 3 is an implementation detail the end user don't have to see. they just want gnome and should get gnome. now we have the situation that the user enables `gnome3` but gets GNOME 40. that is inconsistent and therefore a quality issue

when it's not planned to offer multiple gnome versions, i suggest to just call it `desktopManager.gnome` now instead of changing it to `desktopManager.gnome40`

**Comment by jtojnar** [comment]
Yeah, I wanted to rename the attributes before even starting to work on the update but then I forgot about this. Will need to create ton of aliases. We should probably do that now while we are waiting for someone to double check the fingerprint issue.

**Comment by LEXUGE** [comment]
@jtojnar I tested that out and that was even more serious: after several crashes (restart) in GDM screen, I logged in without my finger or password
Log below:
[log.txt](https://github.com/NixOS/nixpkgs/files/6405387/log.txt)

**Comment by blitz** [comment]
I have tested this on my desktop with AMD graphics. I've seen no issues so far. (I don't have a fingerprint reader, so this is no data point for that issue.)

Thanks for all the work that went into this update!

**Comment by jtojnar** [comment]
@LEXUGE the logs do not say much to me. Could you please open a confidential issue at https://gitlab.gnome.org/GNOME/gdm/-/issues and describe the reproduction steps there? Hopefully, upstream will be able to help us.

**Comment by jonringer** [comment]
since gnome is our primary DM for the graphical iso, I'm going to add this as a release blocker

**Comment by jtojnar** [comment]
@LEXUGE I tried the Fedora workaround patch on my sister’s laptop, and while it still complains, at least it does not seem to be clearing the password field as described on https://gitlab.gnome.org/GNOME/gdm/-/issues/691 any more (I did reproduce it with unpatched 40.0). I managed to log in with fingerprint as well.

Could you verify if it fixes the issue you had?

**Comment by LEXUGE** [comment]
Sure, I will check it out today as soon as possible.

**Comment by jonringer** [comment]
I used the github UI to resolve the merge conflict. If it causes issues, feel to revert / drop it.

Just wanted to start a nixpkgs-review

**Comment by fabianhjr** [comment]
Rebased with master and tried it on my system successfully. No issues so far; didn't even changed my config.

Thanks for the merge request. :+1:

**Review by fabianhjr — APPROVED** [review]
Rebased with master and tried it on my system successfully. No issues so far; didn't even changed my config.

Thanks for the merge request. :+1:

**Comment by jtojnar** [comment]
@LEXUGE oh, sorry. GitHub does not notify of edits, usually.

Thanks for confirming it fixed the password issue. Since it is now possible to log in, I would no longer consider it a blocker and try to debug after this was merged.

For what it is worth, I did manage to log in using fingerprint so it would be nice if you could get a fresh log (with the patches applied) so that we could compare them.

**Pull request merged**
